src/Sarif.Multitool/Rules/SARIF2004.OptimizeFileSize.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+using Microsoft.Json.Pointer;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class OptimizeFileSize : SarifValidationSkimmerBase
+    {
+        /// <summary>
+        /// SARIF2004
+        /// </summary>
+        public override string Id => RuleId.OptimizeFileSize;
+
+        /// <summary>
+        /// Placeholder_SARIF2004_OptimizeFileSize_FullDescription_Text
+        /// </summary>
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString { Text = RuleResources.SARIF2004_OptimizeFileSize_FullDescription_Text };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+                    nameof(RuleResources.SARIF2004_OptimizeFileSize_Warning_EliminateLocationOnlyArtifacts_Text)
+                };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            // We only verify first item in the results and artifacts array,
+            // since tools will typically generate similar nodes.
+            // This approach may cause false negatives.
+            ArtifactLocation firstLocationInArtifactsArray = run.Artifacts[0].Location;
+            ArtifactLocation firstlocationInResults = run.Results[0].Locations[0].PhysicalLocation.ArtifactLocation;
```

Comment:
i am unsure if i have gotten it right. this is what i understood from our convo, so i could be wrong!

Reply:
Yes, that is what we discussed. Careful about possible nulls: `run?.Artifacts?[0]?.Location` and `run?.Results?[0].Locations?[0]?.PhysicalLocation?.ArtifactLocation`.

---
In reply to: [445259975](https://github.com/microsoft/sarif-sdk/pull/1939#discussion_r445259975) [](ancestors = 445259975)